### PR TITLE
Range and heat sensors buff, better range input

### PR DESCRIPTION
**Pull request by DreamySkrell**
image
image
image

---------------
image

### CODE REVIEW

**Comment by DreamySkrell** [comment]
all my homies hate input()

File: code/modules/overmap/ships/computers/sensors.dm [hunk 1]
```diff
@@ -180,6 +183,13 @@
 			if (nrange)
 				sensors.set_range(Clamp(nrange, 1, world.view))
 			return TOPIC_REFRESH
+		if (href_list["range_choice"])
```
Comment (SleepyGemmy) [inline code/modules/overmap/ships/computers/sensors.dm:186]:
```suggestion
		if(href_list["range_choice"])
```

File: code/modules/overmap/ships/computers/sensors.dm [hunk 2]
```diff
@@ -180,6 +183,13 @@
...
+			var/nrange = text2num(href_list["range_choice"])
+			if(!CanInteract(usr, default_state))
+				return TOPIC_NOACTION
+			if (nrange)
```
Comment (SleepyGemmy) [inline code/modules/overmap/ships/computers/sensors.dm:190]:
```suggestion
			if(nrange)
```

File: html/changelogs/DreamySkrell-sensors-436357547.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+# Optional: Remove this file after generating master changelog.  Useful for PR changelogs that won't get used again.
+delete-after: True
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes:
+  - rscadd: "Changes the sensors console range input to a button select like [1][2][3][4][5][6][7]."
```
Comment (SleepyGemmy) [inline html/changelogs/DreamySkrell-sensors-436357547.yml:41]:
```suggestion
  - tweak: "Changes the sensors console range input to a button selection ranging from 1 to 7."
```

**Review by SleepyGemmy — APPROVED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

**Comment by runecap** [comment]
This prevents you from using decimals and that removes the very common "1.5" sensor range people use. It lets you see 2 tiles without gaining heat; change this without touching heat efficiency in sensors and piloting would become more frustrating, not less.

**Comment by Caelphon** [comment]
> This prevents you from using decimals and that removes the very common "1.5" sensor range people use. It lets you see 2 tiles without gaining heat; change this without touching heat efficiency in sensors and piloting would become more frustrating, not less.

This just sounds like an exploit? You're putting in a number that isn't 2, and getting the same result.

**Comment by runecap** [comment]
> This just sounds like an exploit? You're putting in a number that isn't 2, and getting the same result.

It is, but 1.5 for sensors is so common for piloting that removing it would still be a nerf. I think this is also resolvable through small mapping changes, though. The only ships and shuttles that rely on 1.5 range use the 'weak' variant of ship sensors.
Mapping is probably out of the scope of this PR, however.

**Comment by DreamySkrell** [comment]
> This prevents you from using decimals and that removes the very common "1.5" sensor range people use. It lets you see 2 tiles without gaining heat; change this without touching heat efficiency in sensors and piloting would become more frustrating, not less.

I'd say that's an exploit/oversight, and unintentional, yeah.

Also like, imo, currently heat is way too fast/high, and range way too small, so like feel free to change that if you'd like.

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]
Tweak the heat formula to allow for 3 sensor range without heating up.

**Comment by DreamySkrell** [comment]
image

**Comment by DreamySkrell** [comment]
@NonQueueingMatt Changes addressed I think.

**Pull request merged**